portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,46 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
-
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+
+
+/* for my gallery */
+
+var filenames = ["/Images/myimage1.JPG", 
+				"Images/myimage2.JPG",
+				"/Images/IMG2.JPG",
+				"/Images/IMG3.JPG",
+				"/Images/IMG4.jpeg",
+                "/Images/IMG5.jpg",
+                "/Images/IMG6.JPG",
+                "/Images/IMG7.jpeg",
+                "/Images/IMG8.jpeg"];
+
+var currentImage = 0;	
+
+
+
+function update(){
+	var filename = filenames[currentImage];
+	document.getElementById("myImage").src = filename;
+	document.getElementsByClassName("dot").innerHTML = names[currentImage]; 
+	
 }
+
+function next(){
+	currentImage = currentImage + 1;
```

Comment:
The modulo operator can be really useful for things like this. In most languages it's the `%` character. That lets you write lines 41-44 as `currentImage = (current Image + 1) % filenames.length` . It's up to you which one you prefer, I just wanted to let you know about the option.

Reply:
Thanks for this suggestion. It would definitely make the code cleaner.